processor/intervalprocessor/processor.go

```diff
@@ -37,47 +45,52 @@ type Processor struct {
 	histogramLookup    map[identity.Stream]pmetric.HistogramDataPoint
 	expHistogramLookup map[identity.Stream]pmetric.ExponentialHistogramDataPoint
 	summaryLookup      map[identity.Stream]pmetric.SummaryDataPoint
-
-	config *Config
-
-	nextConsumer consumer.Metrics
 }
 
 func newProcessor(config *Config, log *zap.Logger, nextConsumer consumer.Metrics) *Processor {
 	ctx, cancel := context.WithCancel(context.Background())
 
-	return &Processor{
-		ctx:    ctx,
-		cancel: cancel,
-		logger: log,
-
-		stateLock: sync.Mutex{},
-
-		md:                 pmetric.NewMetrics(),
-		rmLookup:           map[identity.Resource]pmetric.ResourceMetrics{},
-		smLookup:           map[identity.Scope]pmetric.ScopeMetrics{},
-		mLookup:            map[identity.Metric]pmetric.Metric{},
-		numberLookup:       map[identity.Stream]pmetric.NumberDataPoint{},
-		histogramLookup:    map[identity.Stream]pmetric.HistogramDataPoint{},
-		expHistogramLookup: map[identity.Stream]pmetric.ExponentialHistogramDataPoint{},
-		summaryLookup:      map[identity.Stream]pmetric.SummaryDataPoint{},
+	numPartitions := int(config.Interval.Seconds())
+
+	partitions := make([]*Partition, numPartitions)
+	for i := range partitions {
+		partitions[i] = &Partition{
+			md:                 pmetric.NewMetrics(),
+			rmLookup:           make(map[identity.Resource]pmetric.ResourceMetrics, 0),
+			smLookup:           make(map[identity.Scope]pmetric.ScopeMetrics, 0),
+			mLookup:            make(map[identity.Metric]pmetric.Metric, 0),
+			numberLookup:       make(map[identity.Stream]pmetric.NumberDataPoint, 0),
+			histogramLookup:    make(map[identity.Stream]pmetric.HistogramDataPoint, 0),
+			expHistogramLookup: make(map[identity.Stream]pmetric.ExponentialHistogramDataPoint, 0),
+			summaryLookup:      make(map[identity.Stream]pmetric.SummaryDataPoint, 0),
+		}
+	}
 
-		config: config,
+	return &Processor{
+		ctx:           ctx,
+		cancel:        cancel,
+		logger:        log,
+		stateLock:     sync.Mutex{},
+		partitions:    partitions,
+		numPartitions: numPartitions,
+		config:        config,
 
 		nextConsumer: nextConsumer,
 	}
 }
 
 func (p *Processor) Start(_ context.Context, _ component.Host) error {
-	exportTicker := time.NewTicker(p.config.Interval)
+	exportTicker := time.NewTicker(time.Second)
```

Comment:
I think this should be `p.config.Interval / 60` instead of hardcoded to 1 second